### ISSUE TITLE
[BUG] Windows Zabbix-agent fails to return success despite package install succeeding

### ISSUE THREAD

**Issue by ephreal** [saltstack/salt#60032]
**Description**

I am attempting to install the 32-bit zabbix-agent package provided from the saltstack windows repository. The state running the install is shown in the setup section.

When I run the state, the output from the state suggests that the package installed but returns a Result of False.  A check of the computer that the state is running on shows the agent to have been installed properly as well. As can be seen in the image below, the state installed Zabbix successfully and started the service.

![zabbix service](https://user-images.githubusercontent.com/14032178/114765006-7e154b00-9d19-11eb-9d18-046f5dfb8cf0.png)

I did see a previous bug report about a repo failing (https://github.com/saltstack/salt/issues/48682), but this appears to be a separate issue (I'm not using a private repo).

**Setup**

Create an SLS file with the following yaml

```
Zabbix Agent (32-bit):
  pkg.installed:
    - pkgs:
      - zabbix-agent: 'latest'
```

**Steps to Reproduce the behavior**

Run the state provided above on a windows 32-bit minion. The output should look similar to the following.

```
test_machine:
----------
          ID: Zabbix Agent (32-bit)
    Function: pkg.installed
      Result: False
     Comment: The following packages failed to install/update: zabbix-agent
     Started: 11:24:20.439861
    Duration: 61300.605 ms
     Changes:   
              ----------
              Zabbix Agent (32-bit):
                  ----------
                  new:
                      5.2.3.2400
                  old:
              zabbix-agent:
                  ----------
                  install status:
                      success
```

**Expected behavior**

The return value should be True. The package has installed properly and the service is running.

**Versions Report**

I wasn't sure how to get the salt version report off of the minion. I did check the minion version, which is also salt 3003.

Salt master version report
```
Salt Version:
          Salt: 3003
 
Dependency Versions:
          cffi: 1.14.5
      cherrypy: Not Installed
      dateutil: Not Installed
     docker-py: Not Installed
         gitdb: 4.0.7
     gitpython: 3.1.14
        Jinja2: 2.11.3
       libgit2: 1.1.0
      M2Crypto: 0.37.1
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.20
      pycrypto: Not Installed
  pycryptodome: 3.9.9
        pygit2: 1.5.0
        Python: 3.9.2 (default, Feb 20 2021, 18:40:11)
  python-gnupg: Not Installed
        PyYAML: 5.4.1
         PyZMQ: 22.0.3
         smmap: 4.0.0
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: arch  
        locale: utf-8
       machine: x86_64
       release: 5.11.11-arch1-1
        system: Linux
       version: Arch Linux
```

**Additional Information**

Despite the package being installed, attempting to uninstall it with salt afterwords also fails

```
salt 'test_machine pkg.remove zabbix-agent
test_machine:
    ----------
    zabbix-agent:
        ----------
        current:
            not installed
```

**Comment by ephreal** [comment]
Apparently hitting enter just submitted this. Let me go through and actually write things out.

**Comment by ephreal** [comment]
Alright, I think that's everything.

**Comment by twangboy** [comment]
I'm not sure the `latest` options works with the `pkgs` argument. A more common state file for a single package would look like this:
```
Zabbix Agent (32-bit):
  pkg.installed:
    - name: zabbix-agent
    - version: latest
```
or
```
zabbix-agent:
  pkg.installed:
    - version: latest
```
or
```
zabbix-agent:
  pkg.installed
```
If you don't specify a version, and the package is not present, the latest will be installed.

**Comment by ephreal** [comment]
Hi twangboy

I took your recommendation there and attempted an install of the zabbix-agent on the computer again with the 3 alternative states you show examples of. Upon running the state, the Return still shows false, despite the install status being successful. For example, when running the state

```
Zabbix Agent (32-bit):
  pkg.installed:
    - name: zabbix-agent
    - version: latest
```

The output is still

```
test_machine:
----------
          ID: Zabbix Agent (32-bit)
    Function: pkg.installed
        Name: zabbix-agent
      Result: False
     Comment: The following packages failed to install/update: zabbix-agent=5.2.3.2400
     Started: 13:49:18.790376
    Duration: 60243.844 ms
     Changes:   
              ----------
              zabbix-agent:
                  ----------
                  install status:
                      success
```

Is there any additional information that I could provide that might help identify what the root cause is?

**Comment by ephreal** [comment]
Hi twangboy,

I have just discovered that the issue lies outside of saltstack and in the saltstack-repo-ng. The zabbix-agent.sls in saltstack-repo-ng is no longer accurate for the 32-bit agent. I'll close this and either open an issue or a pull request for the repo to update the full_name for 32-bit zabbix. It must have been correct at one point because I have used it to install the 32-bit agent with no issues.

I discovered this by additional playing around with the saltstack-repo-ng sls file for the zabbix agent. I noticed that the full_name of the 32-bit zabbix agent did not match the name of the zabbix program in the control panel installed programs list. After changing the name there to match (from `Zabbix Agent` to `Zabbix Agent (32-bit), the 32 bit version of zabbix began to play nicely with saltstack again.

Thank you so much for the time you did spend looking into this!
